--- configure.in.orig	Wed Aug 28 13:15:56 2002
+++ configure.in	Wed Aug 28 13:05:01 2002
@@ -59,11 +59,14 @@
 AM_CONDITIONAL(ANCIENT_AUTOMAKE, test $AUTO_MAKE_VERSION -lt 14)
 AM_CONDITIONAL(OLD_AUTOMAKE, test $AUTO_MAKE_VERSION -lt 15)
 
+
+AM_CONDITIONAL(ENABLE_XMESA_FX, [false])
+
 # set logging default value
 # with_logging=yes
 AC_ARG_WITH(logging, [  --with-logging          Include logging output (default)])
 if test "x$with_logging" = "xno" ; then
-    AC_DEFINE(FG_NDEBUG)
+    AC_DEFINE(FG_NDEBUG,,[fucken])
 fi
 
 dnl specify if we are building with "electric fence"
@@ -200,8 +203,8 @@
     if test "x$ac_cv_lib_MesaGL_glNewList" = "xyes" ; then
 	AC_CHECK_HEADER(GL/fxmesa.h)
 	if test "x$ac_cv_header_GL_fxmesa_h" = "xyes"; then
-	    AC_DEFINE( XMESA )
-            AC_DEFINE( FX )
+	    AC_DEFINE(XMESA,,[fucken])
+            AC_DEFINE(FX,,[fucken])
         fi
     fi
 
@@ -223,7 +226,7 @@
     dnl Win32 libs
 
     echo Win32 specific hacks...
-    AC_DEFINE(WIN32)
+    AC_DEFINE(WIN32,,[fucken])
 
     dnl force a failed check since we will be building under windoze
     AM_CONDITIONAL(ENABLE_XMESA_FX, test "no" = "yes")
@@ -317,7 +320,7 @@
 	sys/stat.h sys/time.h sys/timeb.h unistd.h windows.h winbase.h values.h )
 
 if test "x$ac_cv_header_stdint_h" = "xyes"; then
-    AC_DEFINE( HAVE_STDINT_H )
+    AC_DEFINE( HAVE_STDINT_H,,[fucken] )
 fi
 
 dnl Checks for typedefs, structures, and compiler characteristics.
