diff -urN SimGear-0.0.18.orig/simgear/bucket/Makefile.am SimGear-0.0.18/simgear/bucket/Makefile.am
--- SimGear-0.0.18.orig/simgear/bucket/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/bucket/Makefile.am	Thu May  2 21:44:43 2002
@@ -14,8 +14,5 @@
 # 	$(top_builddir)/bucket/libsgbucket.a \
 # 	$(top_builddir)/misc/libsgmisc.a
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/debug/Makefile.am SimGear-0.0.18/simgear/debug/Makefile.am
--- SimGear-0.0.18.orig/simgear/debug/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/debug/Makefile.am	Thu May  2 21:44:57 2002
@@ -8,8 +8,5 @@
 
 libsgdebug_a_SOURCES = logstream.cxx
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/ephemeris/Makefile.am SimGear-0.0.18/simgear/ephemeris/Makefile.am
--- SimGear-0.0.18.orig/simgear/ephemeris/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/ephemeris/Makefile.am	Thu May  2 21:45:13 2002
@@ -31,8 +31,5 @@
 	uranus.cxx \
 	venus.cxx
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/io/Makefile.am SimGear-0.0.18/simgear/io/Makefile.am
--- SimGear-0.0.18.orig/simgear/io/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/io/Makefile.am	Thu May  2 21:46:25 2002
@@ -20,11 +20,7 @@
 	sg_socket.cxx \
 	sg_socket_udp.cxx 
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
 
 if IS_MINGW
 NETWORK_LIB = -lwsock32
diff -urN SimGear-0.0.18.orig/simgear/magvar/Makefile.am SimGear-0.0.18/simgear/magvar/Makefile.am
--- SimGear-0.0.18.orig/simgear/magvar/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/magvar/Makefile.am	Thu May  2 21:46:41 2002
@@ -14,8 +14,5 @@
 
 testmagvar_LDADD = $(top_builddir)/simgear/magvar/libsgmagvar.a
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/math/Makefile.am SimGear-0.0.18/simgear/math/Makefile.am
--- SimGear-0.0.18.orig/simgear/math/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/math/Makefile.am	Thu May  2 21:46:55 2002
@@ -24,8 +24,5 @@
 	sg_random.c \
 	vector.cxx
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/metar/Makefile.am SimGear-0.0.18/simgear/metar/Makefile.am
--- SimGear-0.0.18.orig/simgear/metar/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/metar/Makefile.am	Thu May  2 21:47:11 2002
@@ -11,8 +11,5 @@
 	MetarStation.cpp \
 	Prtdmetr.cpp Stspack2.cpp Stspack3.cpp
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/misc/Makefile.am SimGear-0.0.18/simgear/misc/Makefile.am
--- SimGear-0.0.18.orig/simgear/misc/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/misc/Makefile.am	Thu May  2 21:47:28 2002
@@ -30,8 +30,5 @@
 props_test_SOURCES = props_test.cxx
 props_test_LDADD = libsgmisc.a ../xml/libsgxml.a ../debug/libsgdebug.a
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/route/Makefile.am SimGear-0.0.18/simgear/route/Makefile.am
--- SimGear-0.0.18.orig/simgear/route/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/route/Makefile.am	Thu May  2 21:47:43 2002
@@ -8,11 +8,7 @@
 	route.cxx \
 	waypoint.cxx
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
 
 noinst_PROGRAMS = waytest routetest
 
diff -urN SimGear-0.0.18.orig/simgear/screen/Makefile.am SimGear-0.0.18/simgear/screen/Makefile.am
--- SimGear-0.0.18.orig/simgear/screen/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/screen/Makefile.am	Thu May  2 21:48:01 2002
@@ -19,8 +19,5 @@
         tr.cxx \
 	win32-printer.h
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/serial/Makefile.am SimGear-0.0.18/simgear/serial/Makefile.am
--- SimGear-0.0.18.orig/simgear/serial/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/serial/Makefile.am	Thu May  2 21:48:50 2002
@@ -14,8 +14,5 @@
 	$(top_builddir)/simgear/serial/libsgserial.a \
 	$(top_builddir)/simgear/debug/libsgdebug.a
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/sky/Makefile.am SimGear-0.0.18/simgear/sky/Makefile.am
--- SimGear-0.0.18.orig/simgear/sky/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/sky/Makefile.am	Thu May  2 21:49:32 2002
@@ -20,8 +20,5 @@
 	sphere.cxx \
 	stars.cxx
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/threads/Makefile.am SimGear-0.0.18/simgear/threads/Makefile.am
--- SimGear-0.0.18.orig/simgear/threads/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/threads/Makefile.am	Thu May  2 21:49:51 2002
@@ -10,8 +10,5 @@
 libsgthreads_a_SOURCES = \
 	SGThread.cxx
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/timing/Makefile.am SimGear-0.0.18/simgear/timing/Makefile.am
--- SimGear-0.0.18.orig/simgear/timing/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/timing/Makefile.am	Thu May  2 21:50:05 2002
@@ -16,8 +16,5 @@
 	timestamp.cxx \
 	timezone.cxx
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/xgl/Makefile.am SimGear-0.0.18/simgear/xgl/Makefile.am
--- SimGear-0.0.18.orig/simgear/xgl/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/xgl/Makefile.am	Thu May  2 21:50:18 2002
@@ -6,8 +6,5 @@
 
 libsgxgl_a_SOURCES = xgl.c xglUtils.c
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+
diff -urN SimGear-0.0.18.orig/simgear/xml/Makefile.am SimGear-0.0.18/simgear/xml/Makefile.am
--- SimGear-0.0.18.orig/simgear/xml/Makefile.am	Thu May  2 21:43:43 2002
+++ SimGear-0.0.18/simgear/xml/Makefile.am	Thu May  2 21:50:31 2002
@@ -18,8 +18,5 @@
 	xmltok.h xmltok.c \
 	xmltok_impl.h
 
-if OLD_AUTOMAKE
-INCLUDES += -I$(top_srcdir)
-else
 INCLUDES = -I$(top_srcdir)
-endif
+

--- hand/Makefile.am~	Wed Apr  3 23:18:54 2002
+++ edited/Makefile.am	Fri May  3 12:42:22 2002
@@ -7,7 +7,7 @@
 	SimGear.dsp \
 	SimGear.dsw
 
-SUBDIRS	= src-libs simgear
+SUBDIRS	= simgear
 
 #
 # Rule to build RPM distribution package


