diff -uNr SimGear-1.9.1.orig/simgear/io/Makefile.am SimGear-1.9.1/simgear/io/Makefile.am
--- SimGear-1.9.1.orig/simgear/io/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/io/Makefile.am	2009-08-11 18:09:26.000000000 +0200
@@ -20,6 +20,14 @@
 	sg_socket.cxx \
 	sg_socket_udp.cxx 
 
+libsgio_la_LIBADD = \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	-lplibnet \
+	-lz
+
 INCLUDES = -I$(top_srcdir)
 
 noinst_PROGRAMS = decode_binobj socktest lowtest tcp_server tcp_client
diff -uNr SimGear-1.9.1.orig/simgear/math/Makefile.am SimGear-1.9.1/simgear/math/Makefile.am
--- SimGear-1.9.1.orig/simgear/math/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/math/Makefile.am	2009-08-11 18:09:26.000000000 +0200
@@ -52,4 +52,8 @@
 	SGGeod.cxx \
 	SGGeodesy.cxx
 
+libsgmath_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/math/Makefile.am.orig SimGear-1.9.1/simgear/math/Makefile.am.orig
--- SimGear-1.9.1.orig/simgear/math/Makefile.am.orig	1970-01-01 01:00:00.000000000 +0100
+++ SimGear-1.9.1/simgear/math/Makefile.am.orig	2009-08-11 18:09:26.000000000 +0200
@@ -0,0 +1,55 @@
+includedir = @includedir@/math
+
+check_PROGRAMS  = SGMathTest SGGeometryTest
+TESTS           = $(check_PROGRAMS)
+
+SGMathTest_SOURCES = SGMathTest.cxx
+SGMathTest_LDADD = libsgmath.la $(base_LIBS)
+
+SGGeometryTest_SOURCES = SGGeometryTest.cxx
+SGGeometryTest_LDADD = libsgmath.la $(base_LIBS)
+
+lib_LTLIBRARIES = libsgmath.la
+
+include_HEADERS = \
+	interpolater.hxx \
+	leastsqs.hxx \
+	point3d.hxx \
+	polar3d.hxx \
+	sg_geodesy.hxx \
+	sg_random.h \
+	sg_types.hxx \
+	vector.hxx \
+	Math.hxx \
+	SGBox.hxx \
+	SGCMath.hxx \
+	SGGeoc.hxx \
+	SGGeod.hxx \
+	SGGeodesy.hxx \
+	SGGeometry.hxx \
+	SGGeometryFwd.hxx \
+	SGIntersect.hxx \
+	SGLimits.hxx \
+	SGLineSegment.hxx \
+	SGMatrix.hxx \
+	SGMath.hxx \
+	SGMathFwd.hxx \
+	SGMisc.hxx \
+	SGPlane.hxx \
+	SGQuat.hxx \
+	SGRay.hxx \
+	SGSphere.hxx \
+	SGTriangle.hxx \
+	SGVec2.hxx \
+	SGVec3.hxx \
+	SGVec4.hxx
+
+libsgmath_la_SOURCES = \
+	interpolater.cxx \
+	leastsqs.cxx \
+	sg_random.c \
+	vector.cxx \
+	SGGeod.cxx \
+	SGGeodesy.cxx
+
+INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/misc/Makefile.am SimGear-1.9.1/simgear/misc/Makefile.am
--- SimGear-1.9.1.orig/simgear/misc/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/misc/Makefile.am	2009-08-11 18:11:32.000000000 +0200
@@ -24,6 +24,14 @@
 	interpolator.cxx \
 	PathOptions.cxx
 
+libsgmisc_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	-lz \
+	-losg \
+	-losgDB
+	
 noinst_PROGRAMS = tabbed_value_test swap_test
 
 tabbed_value_test_SOURCES = tabbed_values_test.cxx
diff -uNr SimGear-1.9.1.orig/simgear/misc/Makefile.am.orig SimGear-1.9.1/simgear/misc/Makefile.am.orig
--- SimGear-1.9.1.orig/simgear/misc/Makefile.am.orig	1970-01-01 01:00:00.000000000 +0100
+++ SimGear-1.9.1/simgear/misc/Makefile.am.orig	2009-08-11 18:09:26.000000000 +0200
@@ -0,0 +1,43 @@
+includedir = @includedir@/misc
+
+lib_LTLIBRARIES = libsgmisc.la
+
+include_HEADERS = \
+	sg_path.hxx \
+	sgstream.hxx \
+	stopwatch.hxx \
+	strutils.hxx \
+	tabbed_values.hxx \
+	texcoord.hxx \
+	zfstream.hxx \
+	interpolator.hxx \
+	stdint.hxx \
+	PathOptions.hxx
+
+libsgmisc_la_SOURCES = \
+	sg_path.cxx \
+	sgstream.cxx \
+	strutils.cxx \
+	tabbed_values.cxx \
+	texcoord.cxx \
+	zfstream.cxx \
+	interpolator.cxx \
+	PathOptions.cxx
+
+noinst_PROGRAMS = tabbed_value_test swap_test
+
+tabbed_value_test_SOURCES = tabbed_values_test.cxx
+tabbed_value_test_LDADD = \
+	libsgmisc.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la
+
+$(top_builddir)/simgear/xml/libsgxml.a:
+	cd $(top_builddir)/simgear/xml && $(MAKE) $(AM_MAKEFLAGS) libsgxml.la
+
+$(top_builddir)/simgear/debug/libsgdebug.a:
+	cd $(top_builddir)/simgear/debug && $(MAKE) $(AM_MAKEFLAGS) libsgdebug.la
+
+swap_test_SOURCES = swap_test.cpp
+
+INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/props/Makefile.am SimGear-1.9.1/simgear/props/Makefile.am
--- SimGear-1.9.1.orig/simgear/props/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/props/Makefile.am	2009-08-11 18:09:26.000000000 +0200
@@ -12,6 +12,8 @@
 	props.cxx \
 	props_io.cxx
 
+libsgprops_la_LIBADD = \
+	$(top_builddir)/simgear/xml/libsgxml.la
 
 # NOTE: Looped dependencies misc and props test files, commenting this one out
 # noinst_PROGRAMS = props_test
diff -uNr SimGear-1.9.1.orig/simgear/route/Makefile.am SimGear-1.9.1/simgear/route/Makefile.am
--- SimGear-1.9.1.orig/simgear/route/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/route/Makefile.am	2009-08-11 18:09:26.000000000 +0200
@@ -8,6 +8,9 @@
 	route.cxx \
 	waypoint.cxx
 
+libsgroute_la_LIBADD = \
+	$(top_builddir)/simgear/math/libsgmath.la
+
 INCLUDES = -I$(top_srcdir)
 
 noinst_PROGRAMS = waytest routetest
diff -uNr SimGear-1.9.1.orig/simgear/screen/Makefile.am SimGear-1.9.1/simgear/screen/Makefile.am
--- SimGear-1.9.1.orig/simgear/screen/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/screen/Makefile.am	2009-08-11 18:09:26.000000000 +0200
@@ -35,6 +35,10 @@
 	shader.cpp \
 	win32-printer.h
 
+libsgscreen_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	-lz -lplibssg
+
 if HAVE_GLUT
 noinst_PROGRAMS = TestRenderTexture
 
diff -uNr SimGear-1.9.1.orig/simgear/sound/Makefile.am SimGear-1.9.1/simgear/sound/Makefile.am
--- SimGear-1.9.1.orig/simgear/sound/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/sound/Makefile.am	2009-08-11 18:09:26.000000000 +0200
@@ -16,6 +16,13 @@
 	soundmgr_openal.cxx \
 	xmlsound.cxx 
 
+libsgsound_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	-lalut
+
 noinst_PROGRAMS = openal_test1 openal_test2
 
 openal_test1_SOURCES = openal_test1.cxx
diff -uNr SimGear-1.9.1.orig/simgear/structure/Makefile.am SimGear-1.9.1/simgear/structure/Makefile.am
--- SimGear-1.9.1.orig/simgear/structure/Makefile.am	2009-08-11 14:39:56.000000000 +0200
+++ SimGear-1.9.1/simgear/structure/Makefile.am	2009-08-11 18:09:26.000000000 +0200
@@ -30,5 +30,9 @@
 	SGSmplhist.cxx \
 	SGSmplstat.cxx
 
+libsgstructure_la_LIBADD = \
+	$(top_builddir)/simgear/timing/libsgtiming.la \
+	-lz
+
 INCLUDES = -I$(top_srcdir)
 
