diff -Naur SimGear-0.3.5/configure.ac SimGear-0.3.5-p/configure.ac
--- SimGear-0.3.5/configure.ac	Fri Mar 26 22:07:15 2004
+++ SimGear-0.3.5-p/configure.ac	Mon Mar 29 11:47:51 2004
@@ -34,7 +34,7 @@
 AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_CXX
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_PROG_INSTALL
 AC_PROG_LN_S
 
diff -Naur SimGear-0.3.5/simgear/Makefile.am SimGear-0.3.5-p/simgear/Makefile.am
--- SimGear-0.3.5/simgear/Makefile.am	Thu Mar 18 15:51:22 2004
+++ SimGear-0.3.5-p/simgear/Makefile.am	Mon Mar 29 11:47:51 2004
@@ -16,10 +16,12 @@
 	$(compatibility_DIR) \
 	xml \
 	debug \
-	misc \
 	structure \
+	timing \
+	misc \
 	bucket \
 	ephemeris \
+	serial \
 	io \
 	magvar \
 	math \
@@ -29,10 +31,8 @@
 	route \
 	scene \
 	screen \
-	serial \
 	sound \
 	$(SGTHREAD_DIR) \
-	timing \
 	xgl
 
 DIST_SUBDIRS = $(SUBDIRS) compatibility threads
diff -Naur SimGear-0.3.5/simgear/bucket/Makefile.am SimGear-0.3.5-p/simgear/bucket/Makefile.am
--- SimGear-0.3.5/simgear/bucket/Makefile.am	Sat Sep  7 04:58:19 2002
+++ SimGear-0.3.5-p/simgear/bucket/Makefile.am	Mon Mar 29 11:47:51 2004
@@ -1,10 +1,10 @@
 includedir = @includedir@/bucket
 
-lib_LIBRARIES = libsgbucket.a
+lib_LTLIBRARIES = libsgbucket.la
 
 include_HEADERS = newbucket.hxx
 
-libsgbucket_a_SOURCES = newbucket.cxx
+libsgbucket_la_SOURCES = newbucket.cxx
 
 # noinst_PROGRAMS = testbucket
 
diff -Naur SimGear-0.3.5/simgear/debug/Makefile.am SimGear-0.3.5-p/simgear/debug/Makefile.am
--- SimGear-0.3.5/simgear/debug/Makefile.am	Sat Sep  7 04:58:19 2002
+++ SimGear-0.3.5-p/simgear/debug/Makefile.am	Mon Mar 29 11:47:51 2004
@@ -2,10 +2,10 @@
 
 EXTRA_DIST = logtest.cxx
 
-lib_LIBRARIES = libsgdebug.a
+lib_LTLIBRARIES = libsgdebug.la
 
 include_HEADERS = debug_types.h logstream.hxx
 
-libsgdebug_a_SOURCES = logstream.cxx
+libsgdebug_la_SOURCES = logstream.cxx
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/environment/Makefile.am SimGear-0.3.5-p/simgear/environment/Makefile.am
--- SimGear-0.3.5/simgear/environment/Makefile.am	Mon Feb  2 11:12:31 2004
+++ SimGear-0.3.5-p/simgear/environment/Makefile.am	Mon Mar 29 11:54:27 2004
@@ -1,9 +1,9 @@
 includedir = @includedir@/environment
 
-lib_LIBRARIES = libsgenvironment.a
+lib_LTLIBRARIES = libsgenvironment.la
 
 include_HEADERS = metar.hxx
 
-libsgenvironment_a_SOURCES = metar.cxx
+libsgenvironment_la_SOURCES = metar.cxx
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/ephemeris/Makefile.am SimGear-0.3.5-p/simgear/ephemeris/Makefile.am
--- SimGear-0.3.5/simgear/ephemeris/Makefile.am	Fri Aug 29 19:36:27 2003
+++ SimGear-0.3.5-p/simgear/ephemeris/Makefile.am	Mon Mar 29 11:47:51 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/ephemeris
 
-lib_LIBRARIES = libsgephem.a
+lib_LTLIBRARIES = libsgephem.la
 
 include_HEADERS = \
 	celestialBody.hxx \
@@ -16,7 +16,7 @@
 	uranus.hxx \
 	venus.hxx
 
-libsgephem_a_SOURCES = \
+libsgephem_la_SOURCES = \
 	celestialBody.cxx \
 	ephemeris.cxx \
 	jupiter.cxx \
diff -Naur SimGear-0.3.5/simgear/io/Makefile.am SimGear-0.3.5-p/simgear/io/Makefile.am
--- SimGear-0.3.5/simgear/io/Makefile.am	Mon Dec 29 16:56:17 2003
+++ SimGear-0.3.5-p/simgear/io/Makefile.am	Mon Mar 29 11:49:03 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/io
 
-lib_LIBRARIES = libsgio.a
+lib_LTLIBRARIES = libsgio.la
 
 include_HEADERS = \
 	iochannel.hxx \
@@ -11,7 +11,7 @@
 	sg_socket.hxx \
 	sg_socket_udp.hxx
 
-libsgio_a_SOURCES = \
+libsgio_la_SOURCES = \
 	iochannel.cxx \
 	lowlevel.cxx \
 	sg_binobj.cxx \
@@ -27,10 +27,11 @@
 tcp_server_SOURCES = tcp_server.cxx
 
 tcp_server_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
 	-lplibnet -lplibul -lz \
 	$(network_LIBS) \
 	$(base_LIBS)
@@ -38,10 +39,11 @@
 tcp_client_SOURCES = tcp_client.cxx
 
 tcp_client_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
 	-lplibnet -lplibul -lz \
 	$(network_LIBS) \
 	$(base_LIBS)
@@ -49,10 +51,10 @@
 socktest_SOURCES = socktest.cxx
 
 socktest_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
 	-lplibnet -lplibul -lz \
 	$(network_LIBS) \
 	$(base_LIBS)
@@ -60,17 +62,17 @@
 lowtest_SOURCES = lowtest.cxx
 
 lowtest_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
 	$(base_LIBS) -lz
 
 decode_binobj_SOURCES = decode_binobj.cxx
 
 decode_binobj_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
 	$(base_LIBS) -lz
diff -Naur SimGear-0.3.5/simgear/magvar/Makefile.am SimGear-0.3.5-p/simgear/magvar/Makefile.am
--- SimGear-0.3.5/simgear/magvar/Makefile.am	Fri Aug 29 19:36:28 2003
+++ SimGear-0.3.5-p/simgear/magvar/Makefile.am	Mon Mar 29 11:47:51 2004
@@ -1,10 +1,10 @@
 includedir = @includedir@/magvar
 
-lib_LIBRARIES = libsgmagvar.a
+lib_LTLIBRARIES = libsgmagvar.la
 
 include_HEADERS = magvar.hxx
 
-libsgmagvar_a_SOURCES = \
+libsgmagvar_la_SOURCES = \
 	coremag.hxx coremag.cxx \
 	magvar.cxx
 
@@ -13,7 +13,7 @@
 testmagvar_SOURCES = testmagvar.cxx
 
 testmagvar_LDADD = \
-	$(top_builddir)/simgear/magvar/libsgmagvar.a \
+	$(top_builddir)/simgear/magvar/libsgmagvar.la \
 	$(base_LIBS)
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/math/Makefile.am SimGear-0.3.5-p/simgear/math/Makefile.am
--- SimGear-0.3.5/simgear/math/Makefile.am	Mon Jun 30 16:54:19 2003
+++ SimGear-0.3.5-p/simgear/math/Makefile.am	Mon Mar 29 11:47:51 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/math
 
-lib_LIBRARIES = libsgmath.a
+lib_LTLIBRARIES = libsgmath.la
 
 include_HEADERS = \
 	interpolater.hxx \
@@ -17,7 +17,7 @@
 
 EXTRA_DIST = linintp2.h linintp2.inl sphrintp.h sphrintp.inl
 
-libsgmath_a_SOURCES = \
+libsgmath_la_SOURCES = \
 	interpolater.cxx \
 	leastsqs.cxx \
 	polar3d.cxx \
diff -Naur SimGear-0.3.5/simgear/misc/Makefile.am SimGear-0.3.5-p/simgear/misc/Makefile.am
--- SimGear-0.3.5/simgear/misc/Makefile.am	Wed Dec 10 17:04:19 2003
+++ SimGear-0.3.5-p/simgear/misc/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/misc
 
-lib_LIBRARIES = libsgmisc.a
+lib_LTLIBRARIES = libsgmisc.la
 
 include_HEADERS = \
 	sg_path.hxx \
@@ -12,7 +12,7 @@
 	zfstream.hxx \
 	interpolator.hxx
 
-libsgmisc_a_SOURCES = \
+libsgmisc_la_SOURCES = \
 	sg_path.cxx \
 	sgstream.cxx \
 	strutils.cxx \
@@ -25,8 +25,11 @@
 
 tabbed_value_test_SOURCES = tabbed_values_test.cxx
 tabbed_value_test_LDADD = \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
-	$(top_builddir)/simgear/xml/libsgxml.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	$(top_builddir)/simgear/timing/libsgtiming.la \
+	-lz
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/nasal/Makefile.am SimGear-0.3.5-p/simgear/nasal/Makefile.am
--- SimGear-0.3.5/simgear/nasal/Makefile.am	Thu Mar 18 15:51:23 2004
+++ SimGear-0.3.5-p/simgear/nasal/Makefile.am	Mon Mar 29 11:49:49 2004
@@ -1,10 +1,10 @@
 includedir = @includedir@/nasal
 
-lib_LIBRARIES = libsgnasal.a
+lib_LTLIBRARIES = libsgnasal.la
 
 include_HEADERS = nasal.h
 
-libsgnasal_a_SOURCES = \
+libsgnasal_la_SOURCES = \
 	code.c code.h \
 	codegen.c \
         data.h \
diff -Naur SimGear-0.3.5/simgear/props/Makefile.am SimGear-0.3.5-p/simgear/props/Makefile.am
--- SimGear-0.3.5/simgear/props/Makefile.am	Tue Sep 30 17:36:51 2003
+++ SimGear-0.3.5-p/simgear/props/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,13 +1,13 @@
 includedir = @includedir@/props
 
-lib_LIBRARIES = libsgprops.a
+lib_LTLIBRARIES = libsgprops.la
 
 include_HEADERS = \
 	condition.hxx \
 	props.hxx \
 	props_io.hxx
 
-libsgprops_a_SOURCES = \
+libsgprops_la_SOURCES = \
 	condition.cxx \
 	props.cxx \
 	props_io.cxx
@@ -16,10 +16,12 @@
 
 props_test_SOURCES = props_test.cxx
 props_test_LDADD = \
-	$(top_builddir)/simgear/props/libsgprops.a \
-	$(top_builddir)/simgear/xml/libsgxml.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/structure/libsgstructure.a
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	$(top_builddir)/simgear/timing/libsgtiming.la \
+	-lz
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/route/Makefile.am SimGear-0.3.5-p/simgear/route/Makefile.am
--- SimGear-0.3.5/simgear/route/Makefile.am	Fri Aug 29 19:36:29 2003
+++ SimGear-0.3.5-p/simgear/route/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,10 +1,10 @@
 includedir = @includedir@/route
 
-lib_LIBRARIES = libsgroute.a
+lib_LTLIBRARIES = libsgroute.la
 
 include_HEADERS = route.hxx waypoint.hxx
 
-libsgroute_a_SOURCES = \
+libsgroute_la_SOURCES = \
 	route.cxx \
 	waypoint.cxx
 
@@ -15,17 +15,19 @@
 waytest_SOURCES = waytest.cxx
 
 waytest_LDADD = \
-	$(top_builddir)/simgear/route/libsgroute.a \
-	$(top_builddir)/simgear/math/libsgmath.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
+	$(top_builddir)/simgear/route/libsgroute.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
 	$(base_LIBS) \
 	-lz
 
 routetest_SOURCES = routetest.cxx
 
 routetest_LDADD = \
-	$(top_builddir)/simgear/route/libsgroute.a \
-	$(top_builddir)/simgear/math/libsgmath.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
+	$(top_builddir)/simgear/route/libsgroute.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	-lz
 	$(base_LIBS)
diff -Naur SimGear-0.3.5/simgear/scene/material/Makefile.am SimGear-0.3.5-p/simgear/scene/material/Makefile.am
--- SimGear-0.3.5/simgear/scene/material/Makefile.am	Fri May 16 15:21:20 2003
+++ SimGear-0.3.5-p/simgear/scene/material/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/scene/material
 
-lib_LIBRARIES = libsgmaterial.a
+lib_LTLIBRARIES = libsgmaterial.la
 
 noinst_HEADERS =
 
@@ -9,7 +9,7 @@
 	matlib.hxx \
 	matmodel.hxx
 
-libsgmaterial_a_SOURCES = \
+libsgmaterial_la_SOURCES = \
 	mat.cxx \
 	matlib.cxx \
 	matmodel.cxx
diff -Naur SimGear-0.3.5/simgear/scene/model/Makefile.am SimGear-0.3.5-p/simgear/scene/model/Makefile.am
--- SimGear-0.3.5/simgear/scene/model/Makefile.am	Fri May 16 15:21:20 2003
+++ SimGear-0.3.5-p/simgear/scene/model/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/scene/model
 
-lib_LIBRARIES = libsgmodel.a
+lib_LTLIBRARIES = libsgmodel.la
 
 noinst_HEADERS =
 
@@ -11,7 +11,7 @@
 	modellib.hxx \
 	placement.hxx
 
-libsgmodel_a_SOURCES = \
+libsgmodel_la_SOURCES = \
 	animation.cxx \
 	location.cxx \
 	model.cxx \
diff -Naur SimGear-0.3.5/simgear/scene/sky/Makefile.am SimGear-0.3.5-p/simgear/scene/sky/Makefile.am
--- SimGear-0.3.5/simgear/scene/sky/Makefile.am	Mon Aug  4 19:45:34 2003
+++ SimGear-0.3.5-p/simgear/scene/sky/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -2,7 +2,7 @@
 
 SUBDIRS = clouds3d
 
-lib_LIBRARIES = libsgsky.a
+lib_LTLIBRARIES = libsgsky.la
 
 include_HEADERS = \
 	cloud.hxx \
@@ -13,7 +13,7 @@
 	sphere.hxx \
 	stars.hxx
 
-libsgsky_a_SOURCES = \
+libsgsky_la_SOURCES = \
 	cloud.cxx \
 	dome.cxx \
 	moon.cxx \
diff -Naur SimGear-0.3.5/simgear/scene/sky/clouds3d/Makefile.am SimGear-0.3.5-p/simgear/scene/sky/clouds3d/Makefile.am
--- SimGear-0.3.5/simgear/scene/sky/clouds3d/Makefile.am	Mon Aug 11 23:17:02 2003
+++ SimGear-0.3.5-p/simgear/scene/sky/clouds3d/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -9,13 +9,13 @@
   EXTGL_SOURCE = 
 endif
 
-lib_LIBRARIES = libsgclouds3d.a
+lib_LTLIBRARIES = libsgclouds3d.la
 
 include_HEADERS = \
 	SkySceneLoader.hpp \
 	SkyUtil.hpp
 
-libsgclouds3d_a_SOURCES = \
+libsgclouds3d_la_SOURCES = \
         $(EXTGL_SOURCE) \
 	vec3fv.cpp vec3fv.hpp vec3f.hpp vec4f.hpp vec2f.hpp \
 	mat16fv.cpp mat16fv.hpp \
diff -Naur SimGear-0.3.5/simgear/scene/tgdb/Makefile.am SimGear-0.3.5-p/simgear/scene/tgdb/Makefile.am
--- SimGear-0.3.5/simgear/scene/tgdb/Makefile.am	Tue Dec 30 20:36:48 2003
+++ SimGear-0.3.5-p/simgear/scene/tgdb/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/scene/tgdb
 
-lib_LIBRARIES = libsgtgdb.a
+lib_LTLIBRARIES = libsgtgdb.la
 
 noinst_HEADERS =
 
@@ -12,7 +12,7 @@
 	userdata.hxx \
 	vasi.hxx
 
-libsgtgdb_a_SOURCES = \
+libsgtgdb_la_SOURCES = \
 	apt_signs.cxx \
 	leaf.cxx \
 	obj.cxx \
diff -Naur SimGear-0.3.5/simgear/screen/Makefile.am SimGear-0.3.5-p/simgear/screen/Makefile.am
--- SimGear-0.3.5/simgear/screen/Makefile.am	Fri Aug 29 19:36:30 2003
+++ SimGear-0.3.5-p/simgear/screen/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/screen
 
-lib_LIBRARIES = libsgscreen.a
+lib_LTLIBRARIES = libsgscreen.la
 
 EXTRA_DIST = jpgfactory.cxx jpgfactory.hxx
 
@@ -22,7 +22,7 @@
 	extensions.hxx \
 	tr.h
 
-libsgscreen_a_SOURCES = \
+libsgscreen_la_SOURCES = \
 	texture.cxx \
 	GLBitmaps.cxx GLBitmaps.h \
 	$(IMAGE_SERVER_SRCS) \
diff -Naur SimGear-0.3.5/simgear/serial/Makefile.am SimGear-0.3.5-p/simgear/serial/Makefile.am
--- SimGear-0.3.5/simgear/serial/Makefile.am	Sat Sep  7 04:58:19 2002
+++ SimGear-0.3.5-p/simgear/serial/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,17 +1,17 @@
 includedir = @includedir@/serial
 
-lib_LIBRARIES = libsgserial.a
+lib_LTLIBRARIES = libsgserial.la
 
 include_HEADERS = serial.hxx
 
-libsgserial_a_SOURCES = serial.cxx
+libsgserial_la_SOURCES = serial.cxx
 
 noinst_PROGRAMS = testserial
 
 testserial_SOURCES = testserial.cxx
 
 testserial_LDADD = \
-	$(top_builddir)/simgear/serial/libsgserial.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/sound/Makefile.am SimGear-0.3.5-p/simgear/sound/Makefile.am
--- SimGear-0.3.5/simgear/sound/Makefile.am	Fri May  9 19:29:07 2003
+++ SimGear-0.3.5-p/simgear/sound/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -2,34 +2,37 @@
 
 EXTRA_DIST = jet.wav
 
-lib_LIBRARIES = libsgsound.a
+lib_LTLIBRARIES = libsgsound.la
 
 noinst_HEADERS =
 
 include_HEADERS = \
 	sample_openal.hxx \
 	soundmgr_openal.hxx \
 	xmlsound.hxx 
 
-libsgsound_a_SOURCES = \
+libsgsound_la_SOURCES = \
 	sample_openal.cxx  \
 	soundmgr_openal.cxx \
 	xmlsound.cxx 
 
 noinst_PROGRAMS = openal_test1 openal_test2
 
 openal_test1_SOURCES = openal_test1.cxx
 openal_test2_SOURCES = openal_test2.cxx
 
 openal_test1_LDADD = \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
 	$(openal_LIBS)
 
 openal_test2_LDADD = \
-	$(top_builddir)/simgear/sound/libsgsound.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
-	$(top_builddir)/simgear/structure/libsgstructure.a \
+	$(top_builddir)/simgear/sound/libsgsound.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	-lz \
 	$(openal_LIBS)
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/structure/Makefile.am SimGear-0.3.5-p/simgear/structure/Makefile.am
--- SimGear-0.3.5/simgear/structure/Makefile.am	Wed Sep 24 19:19:23 2003
+++ SimGear-0.3.5-p/simgear/structure/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/structure
 
-lib_LIBRARIES = libsgstructure.a
+lib_LTLIBRARIES = libsgstructure.la
 
 include_HEADERS = \
 	callback.hxx \
@@ -9,7 +9,7 @@
 	event_mgr.hxx \
 	subsystem_mgr.hxx
 
-libsgstructure_a_SOURCES = \
+libsgstructure_la_SOURCES = \
 	commands.cxx \
 	exception.cxx \
 	event_mgr.cxx\
diff -Naur SimGear-0.3.5/simgear/threads/Makefile.am SimGear-0.3.5-p/simgear/threads/Makefile.am
--- SimGear-0.3.5/simgear/threads/Makefile.am	Sat Sep  7 04:58:20 2002
+++ SimGear-0.3.5-p/simgear/threads/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,13 +1,13 @@
 includedir = @includedir@/threads
 
-lib_LIBRARIES = libsgthreads.a
+lib_LTLIBRARIES = libsgthreads.la
 
 include_HEADERS = \
 	SGGuard.hxx \
 	SGQueue.hxx \
 	SGThread.hxx
 
-libsgthreads_a_SOURCES = \
+libsgthreads_la_SOURCES = \
 	SGThread.cxx
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/timing/Makefile.am SimGear-0.3.5-p/simgear/timing/Makefile.am
--- SimGear-0.3.5/simgear/timing/Makefile.am	Sat Sep  7 04:58:19 2002
+++ SimGear-0.3.5-p/simgear/timing/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/timing
 
-lib_LIBRARIES = libsgtiming.a
+lib_LTLIBRARIES = libsgtiming.la
 
 include_HEADERS = \
 	geocoord.h \
@@ -9,7 +9,7 @@
 	timestamp.hxx \
 	timezone.h
 
-libsgtiming_a_SOURCES = \
+libsgtiming_la_SOURCES = \
 	geocoord.cxx \
 	lowleveltime.cxx \
 	sg_time.cxx \
diff -Naur SimGear-0.3.5/simgear/xgl/Makefile.am SimGear-0.3.5-p/simgear/xgl/Makefile.am
--- SimGear-0.3.5/simgear/xgl/Makefile.am	Sat Sep  7 04:58:19 2002
+++ SimGear-0.3.5-p/simgear/xgl/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,9 +1,9 @@
 includedir = @includedir@/xgl
 
-lib_LIBRARIES = libsgxgl.a
+lib_LTLIBRARIES = libsgxgl.la
 
 include_HEADERS = xgl.h
 
-libsgxgl_a_SOURCES = xgl.c xglUtils.c
+libsgxgl_la_SOURCES = xgl.c xglUtils.c
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/xml/Makefile.am SimGear-0.3.5-p/simgear/xml/Makefile.am
--- SimGear-0.3.5/simgear/xml/Makefile.am	Fri Aug 29 19:36:30 2003
+++ SimGear-0.3.5-p/simgear/xml/Makefile.am	Mon Mar 29 11:47:56 2004
@@ -1,6 +1,6 @@
 includedir = @includedir@/xml
 
-lib_LIBRARIES = libsgxml.a
+lib_LTLIBRARIES = libsgxml.la
 
 include_HEADERS = \
 	easyxml.hxx
@@ -8,7 +8,7 @@
 noinst_HEADERS = \
 	xmltok_impl.c xmltok_ns.c
 
-libsgxml_a_SOURCES = \
+libsgxml_la_SOURCES = \
 	asciitab.h \
 	easyxml.cxx \
 	hashtable.h hashtable.c \
