diff -uNr SimGear-1.9.1.orig/simgear/bucket/Makefile.am SimGear-1.9.1/simgear/bucket/Makefile.am
--- SimGear-1.9.1.orig/simgear/bucket/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/bucket/Makefile.am	2009-08-12 11:13:26.000000000 +0200
@@ -5,6 +5,8 @@
 include_HEADERS = newbucket.hxx
 
 libsgbucket_la_SOURCES = newbucket.cxx
+libsgbucket_la_LIBADD = \
+ 	$(top_builddir)/simgear/misc/libsgmisc.la
 
 # noinst_PROGRAMS = testbucket
 
diff -uNr SimGear-1.9.1.orig/simgear/environment/Makefile.am SimGear-1.9.1/simgear/environment/Makefile.am
--- SimGear-1.9.1.orig/simgear/environment/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/environment/Makefile.am	2009-08-12 13:25:18.000000000 +0200
@@ -6,4 +6,13 @@
 
 libsgenvironment_la_SOURCES = metar.cxx visual_enviro.cxx precipitation.cxx
 
+libsgenvironment_la_LIBADD = \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/scene/sky/libsgsky.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	-losg \
+	-losgParticle
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/ephemeris/Makefile.am SimGear-1.9.1/simgear/ephemeris/Makefile.am
--- SimGear-1.9.1.orig/simgear/ephemeris/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/ephemeris/Makefile.am	2009-08-12 12:30:32.000000000 +0200
@@ -31,4 +31,8 @@
 	uranus.cxx \
 	venus.cxx
 
+libsgephem_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/io/Makefile.am SimGear-1.9.1/simgear/io/Makefile.am
--- SimGear-1.9.1.orig/simgear/io/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/io/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -20,6 +20,14 @@
 	sg_socket.cxx \
 	sg_socket_udp.cxx 
 
+libsgio_la_LIBADD = \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	-lplibnet \
+	-lz
+
 INCLUDES = -I$(top_srcdir)
 
 noinst_PROGRAMS = decode_binobj socktest lowtest tcp_server tcp_client
diff -uNr SimGear-1.9.1.orig/simgear/math/Makefile.am SimGear-1.9.1/simgear/math/Makefile.am
--- SimGear-1.9.1.orig/simgear/math/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/math/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -52,4 +52,8 @@
 	SGGeod.cxx \
 	SGGeodesy.cxx
 
+libsgmath_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/misc/Makefile.am SimGear-1.9.1/simgear/misc/Makefile.am
--- SimGear-1.9.1.orig/simgear/misc/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/misc/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -24,6 +24,14 @@
 	interpolator.cxx \
 	PathOptions.cxx
 
+libsgmisc_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	-lz \
+	-losg \
+	-losgDB
+	
 noinst_PROGRAMS = tabbed_value_test swap_test
 
 tabbed_value_test_SOURCES = tabbed_values_test.cxx
diff -uNr SimGear-1.9.1.orig/simgear/nasal/Makefile.am SimGear-1.9.1/simgear/nasal/Makefile.am
--- SimGear-1.9.1.orig/simgear/nasal/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/nasal/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -10,4 +10,8 @@
                        thread-posix.c thread-win32.c threadlib.c	\
                        utf8lib.c vector.c
 
+libsgnasal_la_LIBADD = \
+	-lm \
+	-lpthread
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/props/Makefile.am SimGear-1.9.1/simgear/props/Makefile.am
--- SimGear-1.9.1.orig/simgear/props/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/props/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -12,6 +12,10 @@
 	props.cxx \
 	props_io.cxx
 
+libsgprops_la_LIBADD = \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/timing/libsgtiming.la
 
 # NOTE: Looped dependencies misc and props test files, commenting this one out
 # noinst_PROGRAMS = props_test
diff -uNr SimGear-1.9.1.orig/simgear/route/Makefile.am SimGear-1.9.1/simgear/route/Makefile.am
--- SimGear-1.9.1.orig/simgear/route/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/route/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -8,6 +8,9 @@
 	route.cxx \
 	waypoint.cxx
 
+libsgroute_la_LIBADD = \
+	$(top_builddir)/simgear/math/libsgmath.la
+
 INCLUDES = -I$(top_srcdir)
 
 noinst_PROGRAMS = waytest routetest
diff -uNr SimGear-1.9.1.orig/simgear/scene/Makefile.am SimGear-1.9.1/simgear/scene/Makefile.am
--- SimGear-1.9.1.orig/simgear/scene/Makefile.am	2009-01-25 13:00:32.000000000 +0100
+++ SimGear-1.9.1/simgear/scene/Makefile.am	2009-08-12 14:33:12.000000000 +0200
@@ -1,7 +1,6 @@
 includedir = @includedir@/scene
 
-SUBDIRS = material model sky tgdb util
-
+SUBDIRS = util tgdb model material sky 
 # lib_LIBRARIES = libsgscene.a
 
 # noinst_HEADERS =
diff -uNr SimGear-1.9.1.orig/simgear/scene/material/Makefile.am SimGear-1.9.1/simgear/scene/material/Makefile.am
--- SimGear-1.9.1.orig/simgear/scene/material/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/scene/material/Makefile.am	2009-08-12 13:49:46.000000000 +0200
@@ -14,4 +14,15 @@
 	matlib.cxx \
 	matmodel.cxx
 
+libsgmaterial_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/scene/model/libsgmodel.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	-losg \
+	-losgDB \
+	-lplibul
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/scene/model/Makefile.am SimGear-1.9.1/simgear/scene/model/Makefile.am
--- SimGear-1.9.1.orig/simgear/scene/model/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/scene/model/Makefile.am	2009-08-12 13:29:53.000000000 +0200
@@ -46,4 +46,19 @@
 	SGScaleTransform.cxx \
 	SGTranslateTransform.cxx
 
+libsgmodel_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/screen/libsgscreen.la \
+	$(top_builddir)/simgear/scene/tgdb/libsgtgdb.la \
+	$(top_builddir)/simgear/scene/util/libsgutil.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	-lGL \
+	-losg \
+	-losgDB \
+	-losgParticle \
+	-losgUtil
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/scene/sky/Makefile.am SimGear-1.9.1/simgear/scene/sky/Makefile.am
--- SimGear-1.9.1.orig/simgear/scene/sky/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/scene/sky/Makefile.am	2009-08-12 13:47:18.000000000 +0200
@@ -28,4 +28,17 @@
 	newcloud.cxx \
 	CloudShaderGeometry.cxx
 
+libsgsky_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/scene/model/libsgmodel.la \
+	$(top_builddir)/simgear/scene/util/libsgutil.la \
+	$(top_builddir)/simgear/screen/libsgscreen.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	-losg \
+	-losgDB \
+	-lGL \
+	-lGLU
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/scene/tgdb/Makefile.am SimGear-1.9.1/simgear/scene/tgdb/Makefile.am
--- SimGear-1.9.1.orig/simgear/scene/tgdb/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/scene/tgdb/Makefile.am	2009-08-12 15:05:23.000000000 +0200
@@ -40,4 +39,21 @@
 	TileEntry.cxx \
 	TreeBin.cxx TreeBin.hxx
 
+libsgtgdb_la_LIBADD = \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	-lGL \
+	-losg \
+	-losgDB \
+	-losgUtil
+
+#	$(top_builddir)/scene/util/libsutil.la
+#	$(top_builddir)/scene/model/libsgmodel.la
+#	$(top_builddir)/scene/material/libsgmaterial.la
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/scene/util/Makefile.am SimGear-1.9.1/simgear/scene/util/Makefile.am
--- SimGear-1.9.1.orig/simgear/scene/util/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/scene/util/Makefile.am	2009-08-12 11:55:02.000000000 +0200
@@ -31,4 +31,8 @@
 	StateAttributeFactory.cxx \
 	QuadTreeBuilder.cxx
 
+libsgutil_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	-losg
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/screen/Makefile.am SimGear-1.9.1/simgear/screen/Makefile.am
--- SimGear-1.9.1.orig/simgear/screen/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/screen/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -35,6 +35,12 @@
 	shader.cpp \
 	win32-printer.h
 
+libsgscreen_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	-lz \
+	-lplibssg \
+	-lGLU
+
 if HAVE_GLUT
 noinst_PROGRAMS = TestRenderTexture
 
diff -uNr SimGear-1.9.1.orig/simgear/serial/Makefile.am SimGear-1.9.1/simgear/serial/Makefile.am
--- SimGear-1.9.1.orig/simgear/serial/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/serial/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -6,6 +6,9 @@
 
 libsgserial_la_SOURCES = serial.cxx
 
+libsgserial_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la
+
 noinst_PROGRAMS = testserial
 
 testserial_SOURCES = testserial.cxx
diff -uNr SimGear-1.9.1.orig/simgear/sound/Makefile.am SimGear-1.9.1/simgear/sound/Makefile.am
--- SimGear-1.9.1.orig/simgear/sound/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/sound/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -16,6 +16,13 @@
 	soundmgr_openal.cxx \
 	xmlsound.cxx 
 
+libsgsound_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	-lalut
+
 noinst_PROGRAMS = openal_test1 openal_test2
 
 openal_test1_SOURCES = openal_test1.cxx
diff -uNr SimGear-1.9.1.orig/simgear/structure/Makefile.am SimGear-1.9.1/simgear/structure/Makefile.am
--- SimGear-1.9.1.orig/simgear/structure/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/structure/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -30,5 +30,10 @@
 	SGSmplhist.cxx \
 	SGSmplstat.cxx
 
+libsgstructure_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/timing/libsgtiming.la \
+	-lz
+
 INCLUDES = -I$(top_srcdir)
 
diff -uNr SimGear-1.9.1.orig/simgear/timing/Makefile.am SimGear-1.9.1/simgear/timing/Makefile.am
--- SimGear-1.9.1.orig/simgear/timing/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/timing/Makefile.am	2009-08-12 09:40:02.000000000 +0200
@@ -16,4 +16,7 @@
 	timestamp.cxx \
 	timezone.cxx
 
+libsgtiming_la_LIBADD = \
+	$(top_builddir)/simgear/debug/libsgdebug.la
+
 INCLUDES = -I$(top_srcdir)
diff -uNr SimGear-1.9.1.orig/simgear/xml/Makefile.am SimGear-1.9.1/simgear/xml/Makefile.am
--- SimGear-1.9.1.orig/simgear/xml/Makefile.am	2009-08-12 15:08:50.000000000 +0200
+++ SimGear-1.9.1/simgear/xml/Makefile.am	2009-08-12 09:51:44.000000000 +0200
@@ -18,4 +18,7 @@
 	xmltok.h xmltok.c \
 	xmltok_impl.h
 
+libsgxml_la_LIBADD = \
+	$(top_builddir)/simgear/structure/libsgstructure.la
+
 INCLUDES = -I$(top_srcdir)
