diff -Naur SimGear-0.2.0/configure.ac SimGear-0.2.0-p/configure.ac
--- SimGear-0.2.0/configure.ac	Sat Sep  7 03:25:02 2002
+++ SimGear-0.2.0-p/configure.ac	Fri Sep 13 08:12:18 2002
@@ -34,7 +34,7 @@
 AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_CXX
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_PROG_INSTALL
 AC_PROG_LN_S
 
diff -Naur SimGear-0.2.0/simgear/Makefile.am SimGear-0.2.0-p/simgear/Makefile.am
--- SimGear-0.2.0/simgear/Makefile.am	Mon Sep  2 05:42:51 2002
+++ SimGear-0.2.0-p/simgear/Makefile.am	Fri Sep 13 08:21:08 2002
@@ -18,13 +18,13 @@
 	misc \
 	bucket \
 	ephemeris \
+	serial \
 	io \
 	magvar \
 	math \
 	$(METAR_DIRS) \
 	route \
 	screen \
-	serial \
 	sky \
 	$(SGTHREAD_DIR) \
 	timing \
diff -Naur SimGear-0.2.0/simgear/bucket/Makefile.am SimGear-0.2.0-p/simgear/bucket/Makefile.am
--- SimGear-0.2.0/simgear/bucket/Makefile.am	Sun Aug 25 19:24:15 2002
+++ SimGear-0.2.0-p/simgear/bucket/Makefile.am	Fri Sep 13 08:13:01 2002
@@ -1,10 +1,10 @@
 includedir = @includedir@/bucket
 
-lib_LIBRARIES = libsgbucket.a
+lib_LTLIBRARIES = libsgbucket.la
 
 include_HEADERS = newbucket.hxx
 
-libsgbucket_a_SOURCES = newbucket.cxx
+libsgbucket_la_SOURCES = newbucket.cxx
 
 # noinst_PROGRAMS = testbucket
 
diff -Naur SimGear-0.2.0/simgear/debug/Makefile.am SimGear-0.2.0-p/simgear/debug/Makefile.am
--- SimGear-0.2.0/simgear/debug/Makefile.am	Sun Aug 25 19:24:20 2002
+++ SimGear-0.2.0-p/simgear/debug/Makefile.am	Fri Sep 13 08:13:24 2002
@@ -2,10 +2,10 @@
 
 EXTRA_DIST = logtest.cxx
 
-lib_LIBRARIES = libsgdebug.a
+lib_LTLIBRARIES = libsgdebug.la
 
 include_HEADERS = debug_types.h logstream.hxx
 
-libsgdebug_a_SOURCES = logstream.cxx
+libsgdebug_la_SOURCES = logstream.cxx
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.2.0/simgear/ephemeris/Makefile.am SimGear-0.2.0-p/simgear/ephemeris/Makefile.am
--- SimGear-0.2.0/simgear/ephemeris/Makefile.am	Sun Aug 25 19:24:27 2002
+++ SimGear-0.2.0-p/simgear/ephemeris/Makefile.am	Fri Sep 13 08:13:39 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/ephemeris
 
-lib_LIBRARIES = libsgephem.a
+lib_LTLIBRARIES = libsgephem.la
 
 include_HEADERS = \
 	celestialBody.hxx \
@@ -16,7 +16,7 @@
 	uranus.hxx \
 	venus.hxx
 
-libsgephem_a_SOURCES = \
+libsgephem_la_SOURCES = \
 	celestialBody.cxx \
 	ephemeris.cxx \
 	jupiter.cxx \
diff -Naur SimGear-0.2.0/simgear/io/Makefile.am SimGear-0.2.0-p/simgear/io/Makefile.am
--- SimGear-0.2.0/simgear/io/Makefile.am	Sat Sep  7 01:13:31 2002
+++ SimGear-0.2.0-p/simgear/io/Makefile.am	Fri Sep 13 08:15:47 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/io
 
-lib_LIBRARIES = libsgio.a
+lib_LTLIBRARIES = libsgio.la
 
 include_HEADERS = \
 	iochannel.hxx \
@@ -11,7 +11,7 @@
 	sg_socket.hxx \
 	sg_socket_udp.hxx
 
-libsgio_a_SOURCES = \
+libsgio_la_SOURCES = \
 	iochannel.cxx \
 	lowlevel.cxx \
 	sg_binobj.cxx \
@@ -53,24 +53,32 @@
 socktest_SOURCES = socktest.cxx
 
 socktest_LDADD = \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-        $(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
-	$(top_builddir)/simgear/xml/libsgxml.a \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+        $(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
 	-lplibnet -lplibul -lz $(NETWORK_LIB)
 
 lowtest_SOURCES = lowtest.cxx
 
 lowtest_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	-lplibnet -lplibul -lz $(NETWORK_LIB)
 
 decode_binobj_SOURCES = decode_binobj.cxx
 
 decode_binobj_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-        $(top_builddir)/simgear/bucket/libsgbucket.a \
-        $(top_builddir)/simgear/misc/libsgmisc.a \
-        $(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/xml/libsgxml.a \
-	-lz
+	$(top_builddir)/simgear/io/libsgio.la \
+        $(top_builddir)/simgear/bucket/libsgbucket.la \
+        $(top_builddir)/simgear/misc/libsgmisc.la \
+        $(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(NETWORK_LIB) -lz -lplibnet
diff -Naur SimGear-0.2.0/simgear/magvar/Makefile.am SimGear-0.2.0-p/simgear/magvar/Makefile.am
--- SimGear-0.2.0/simgear/magvar/Makefile.am	Sun Aug 25 19:24:53 2002
+++ SimGear-0.2.0-p/simgear/magvar/Makefile.am	Fri Sep 13 08:16:10 2002
@@ -1,10 +1,10 @@
 includedir = @includedir@/magvar
 
-lib_LIBRARIES = libsgmagvar.a
+lib_LTLIBRARIES = libsgmagvar.la
 
 include_HEADERS = magvar.hxx
 
-libsgmagvar_a_SOURCES = \
+libsgmagvar_la_SOURCES = \
 	coremag.hxx coremag.cxx \
 	magvar.cxx
 
@@ -12,6 +12,6 @@
 
 testmagvar_SOURCES = testmagvar.cxx
 
-testmagvar_LDADD = $(top_builddir)/simgear/magvar/libsgmagvar.a
+testmagvar_LDADD = $(top_builddir)/simgear/magvar/libsgmagvar.la
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.2.0/simgear/math/Makefile.am SimGear-0.2.0-p/simgear/math/Makefile.am
--- SimGear-0.2.0/simgear/math/Makefile.am	Sun Aug 25 19:25:01 2002
+++ SimGear-0.2.0-p/simgear/math/Makefile.am	Fri Sep 13 08:16:39 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/math
 
-lib_LIBRARIES = libsgmath.a
+lib_LTLIBRARIES = libsgmath.la
 
 include_HEADERS = \
 	interpolater.hxx \
@@ -16,7 +16,7 @@
 
 EXTRA_DIST = linintp2.h linintp2.inl sphrintp.h sphrintp.inl
 
-libsgmath_a_SOURCES = \
+libsgmath_la_SOURCES = \
 	interpolater.cxx \
 	leastsqs.cxx \
 	polar3d.cxx \
diff -Naur SimGear-0.2.0/simgear/metar/Makefile.am SimGear-0.2.0-p/simgear/metar/Makefile.am
--- SimGear-0.2.0/simgear/metar/Makefile.am	Sun Aug 25 19:25:07 2002
+++ SimGear-0.2.0-p/simgear/metar/Makefile.am	Fri Sep 13 08:16:56 2002
@@ -1,10 +1,10 @@
 includedir = @includedir@/metar
 
-lib_LIBRARIES = libsgmetar.a
+lib_LTLIBRARIES = libsgmetar.la
 
 include_HEADERS = MetarReport.h MetarStation.h 
 
-libsgmetar_a_SOURCES = \
+libsgmetar_la_SOURCES = \
 	Antoi.cpp Charcmp.cpp Dcdmetar.cpp Dcdmtrmk.cpp Drvmetar.c \
 	Fracpart.cpp Local.h Metar.h \
 	MetarReport.cpp \
diff -Naur SimGear-0.2.0/simgear/misc/Makefile.am SimGear-0.2.0-p/simgear/misc/Makefile.am
--- SimGear-0.2.0/simgear/misc/Makefile.am	Sun Aug 25 19:25:13 2002
+++ SimGear-0.2.0-p/simgear/misc/Makefile.am	Fri Sep 13 08:17:36 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/misc
 
-lib_LIBRARIES = libsgmisc.a
+lib_LTLIBRARIES = libsgmisc.la
 
 include_HEADERS = \
         commands.hxx \
@@ -14,7 +14,7 @@
 	texcoord.hxx \
 	zfstream.hxx
 
-libsgmisc_a_SOURCES = \
+libsgmisc_la_SOURCES = \
         commands.cxx \
         exception.cxx \
 	props.cxx \
@@ -28,6 +28,6 @@
 noinst_PROGRAMS = props_test
 
 props_test_SOURCES = props_test.cxx
-props_test_LDADD = libsgmisc.a ../xml/libsgxml.a ../debug/libsgdebug.a
+props_test_LDADD = libsgmisc.la ../xml/libsgxml.la ../debug/libsgdebug.la -lz
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.2.0/simgear/route/Makefile.am SimGear-0.2.0-p/simgear/route/Makefile.am
--- SimGear-0.2.0/simgear/route/Makefile.am	Sun Aug 25 19:25:18 2002
+++ SimGear-0.2.0-p/simgear/route/Makefile.am	Fri Sep 13 08:18:32 2002
@@ -1,10 +1,10 @@
 includedir = @includedir@/route
 
-lib_LIBRARIES = libsgroute.a
+lib_LTLIBRARIES = libsgroute.la
 
 include_HEADERS = route.hxx waypoint.hxx
 
-libsgroute_a_SOURCES = \
+libsgroute_la_SOURCES = \
 	route.cxx \
 	waypoint.cxx
 
@@ -15,16 +15,19 @@
 waytest_SOURCES = waytest.cxx
 
 waytest_LDADD = \
-	$(top_builddir)/simgear/route/libsgroute.a \
-	$(top_builddir)/simgear/math/libsgmath.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
-	$(top_builddir)/simgear/xml/libsgxml.a \
+	$(top_builddir)/simgear/route/libsgroute.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
 	-lz
 
 routetest_SOURCES = routetest.cxx
 
 routetest_LDADD = \
-	$(top_builddir)/simgear/route/libsgroute.a \
-	$(top_builddir)/simgear/math/libsgmath.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a
+	$(top_builddir)/simgear/route/libsgroute.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	-lz
diff -Naur SimGear-0.2.0/simgear/screen/Makefile.am SimGear-0.2.0-p/simgear/screen/Makefile.am
--- SimGear-0.2.0/simgear/screen/Makefile.am	Sun Aug 25 19:25:25 2002
+++ SimGear-0.2.0-p/simgear/screen/Makefile.am	Fri Sep 13 08:18:51 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/screen
 
-lib_LIBRARIES = libsgscreen.a
+lib_LTLIBRARIES = libsgscreen.la
 
 if ENABLE_JPEG_SERVER
 IMAGE_SERVER_INCL = jpgfactory.hxx
@@ -12,7 +12,7 @@
 
 include_HEADERS = $(IMAGE_SERVER_INCL) screen-dump.hxx tr.h
 
-libsgscreen_a_SOURCES = \
+libsgscreen_la_SOURCES = \
 	GLBitmaps.cxx GLBitmaps.h \
 	$(IMAGE_SERVER_SRCS) \
 	screen-dump.cxx \
diff -Naur SimGear-0.2.0/simgear/serial/Makefile.am SimGear-0.2.0-p/simgear/serial/Makefile.am
--- SimGear-0.2.0/simgear/serial/Makefile.am	Sat Sep  7 02:40:38 2002
+++ SimGear-0.2.0-p/simgear/serial/Makefile.am	Fri Sep 13 08:19:16 2002
@@ -1,17 +1,17 @@
 includedir = @includedir@/serial
 
-lib_LIBRARIES = libsgserial.a
+lib_LTLIBRARIES = libsgserial.la
 
 include_HEADERS = serial.hxx
 
-libsgserial_a_SOURCES = serial.cxx
+libsgserial_la_SOURCES = serial.cxx
 
 noinst_PROGRAMS = testserial
 
 testserial_SOURCES = testserial.cxx
 
 testserial_LDADD = \
-	$(top_builddir)/simgear/serial/libsgserial.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.2.0/simgear/sky/Makefile.am SimGear-0.2.0-p/simgear/sky/Makefile.am
--- SimGear-0.2.0/simgear/sky/Makefile.am	Sun Aug 25 19:25:35 2002
+++ SimGear-0.2.0-p/simgear/sky/Makefile.am	Fri Sep 13 08:19:31 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/sky
 
-lib_LIBRARIES = libsgsky.a
+lib_LTLIBRARIES = libsgsky.la
 
 include_HEADERS = \
 	cloud.hxx \
@@ -11,7 +11,7 @@
 	sphere.hxx \
 	stars.hxx
 
-libsgsky_a_SOURCES = \
+libsgsky_la_SOURCES = \
 	cloud.cxx \
 	dome.cxx \
 	moon.cxx \
diff -Naur SimGear-0.2.0/simgear/threads/Makefile.am SimGear-0.2.0-p/simgear/threads/Makefile.am
--- SimGear-0.2.0/simgear/threads/Makefile.am	Sun Aug 25 19:25:39 2002
+++ SimGear-0.2.0-p/simgear/threads/Makefile.am	Fri Sep 13 08:20:32 2002
@@ -1,13 +1,13 @@
 includedir = @includedir@/threads
 
-lib_LIBRARIES = libsgthreads.a
+lib_LTLIBRARIES = libsgthreads.la
 
 include_HEADERS = \
 	SGGuard.hxx \
 	SGQueue.hxx \
 	SGThread.hxx
 
-libsgthreads_a_SOURCES = \
+libsgthreads_la_SOURCES = \
 	SGThread.cxx
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.2.0/simgear/timing/Makefile.am SimGear-0.2.0-p/simgear/timing/Makefile.am
--- SimGear-0.2.0/simgear/timing/Makefile.am	Sun Aug 25 19:25:43 2002
+++ SimGear-0.2.0-p/simgear/timing/Makefile.am	Fri Sep 13 08:19:41 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/timing
 
-lib_LIBRARIES = libsgtiming.a
+lib_LTLIBRARIES = libsgtiming.la
 
 include_HEADERS = \
 	geocoord.h \
@@ -9,7 +9,7 @@
 	timestamp.hxx \
 	timezone.h
 
-libsgtiming_a_SOURCES = \
+libsgtiming_la_SOURCES = \
 	geocoord.cxx \
 	lowleveltime.cxx \
 	sg_time.cxx \
diff -Naur SimGear-0.2.0/simgear/xgl/Makefile.am SimGear-0.2.0-p/simgear/xgl/Makefile.am
--- SimGear-0.2.0/simgear/xgl/Makefile.am	Sun Aug 25 19:25:47 2002
+++ SimGear-0.2.0-p/simgear/xgl/Makefile.am	Fri Sep 13 08:20:05 2002
@@ -1,9 +1,9 @@
 includedir = @includedir@/xgl
 
-lib_LIBRARIES = libsgxgl.a
+lib_LTLIBRARIES = libsgxgl.la
 
 include_HEADERS = xgl.h
 
-libsgxgl_a_SOURCES = xgl.c xglUtils.c
+libsgxgl_la_SOURCES = xgl.c xglUtils.c
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.2.0/simgear/xml/Makefile.am SimGear-0.2.0-p/simgear/xml/Makefile.am
--- SimGear-0.2.0/simgear/xml/Makefile.am	Sun Aug 25 19:25:52 2002
+++ SimGear-0.2.0-p/simgear/xml/Makefile.am	Fri Sep 13 08:20:15 2002
@@ -1,6 +1,6 @@
 includedir = @includedir@/xml
 
-lib_LIBRARIES = libsgxml.a
+lib_LTLIBRARIES = libsgxml.la
 
 include_HEADERS = \
 	easyxml.hxx
@@ -8,7 +8,7 @@
 noinst_HEADERS = \
 	xmltok_impl.c xmltok_ns.c
 
-libsgxml_a_SOURCES = \
+libsgxml_la_SOURCES = \
         asciitab.h \
 	easyxml.cxx \
 	hashtable.h hashtable.c \
diff -Naur SimGear-0.2.0/simgear/io/Makefile.am SimGear-0.2.0-p/simgear/io/Makefile.am
--- SimGear-0.2.0/simgear/io/Makefile.am	Fri Sep 13 20:44:42 2002
+++ SimGear-0.2.0-p/simgear/io/Makefile.am	Fri Sep 13 20:45:21 2002
@@ -33,21 +33,23 @@
 tcp_server_SOURCES = tcp_server.cxx
 
 tcp_server_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
-	$(top_builddir)/simgear/xml/libsgxml.a \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
 	-lplibnet -lplibul -lz $(NETWORK_LIB)
 
 tcp_client_SOURCES = tcp_client.cxx
 
 tcp_client_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a \
-	$(top_builddir)/simgear/bucket/libsgbucket.a \
-	$(top_builddir)/simgear/misc/libsgmisc.a \
-        $(top_builddir)/simgear/xml/libsgxml.a \
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/bucket/libsgbucket.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la \
+        $(top_builddir)/simgear/xml/libsgxml.la \
+        $(top_builddir)/simgear/serial/libsgserial.la \
 	-lplibnet -lplibul -lz $(NETWORK_LIB)
 
 socktest_SOURCES = socktest.cxx
diff -Naur SimGear-0.3.1/simgear/sky/clouds3d/Makefile.am SimGear-0.3.1-p/simgear/sky/clouds3d/Makefile.am
--- SimGear-0.3.1/simgear/sky/clouds3d/Makefile.am	Wed Dec  4 20:34:36 2002
+++ SimGear-0.3.1-p/simgear/sky/clouds3d/Makefile.am	Sun May  4 18:44:37 2003
@@ -9,13 +9,13 @@
   EXTGL_SOURCE = 
 endif
 
-lib_LIBRARIES = libsgclouds3d.a
+lib_LTLIBRARIES = libsgclouds3d.la
 
 include_HEADERS = \
 	SkySceneLoader.hpp \
 	SkyUtil.hpp
 
-libsgclouds3d_a_SOURCES = \
+libsgclouds3d_la_SOURCES = \
         $(EXTGL_SOURCE) \
 	vec3fv.cpp vec3fv.hpp vec3f.hpp vec4f.hpp vec2f.hpp \
 	mat16fv.cpp mat16fv.hpp \
