diff -Naur SimGear-0.3.5/simgear/misc/Makefile.am SimGear-0.3.5-p/simgear/misc/Makefile.am
--- SimGear-0.3.5/simgear/misc/Makefile.am	Mon Mar 29 12:18:56 2004
+++ SimGear-0.3.5-p/simgear/misc/Makefile.am	Mon Mar 29 12:19:35 2004
@@ -30,6 +30,7 @@
 	$(top_builddir)/simgear/debug/libsgdebug.la \
 	$(top_builddir)/simgear/structure/libsgstructure.la \
 	$(top_builddir)/simgear/timing/libsgtiming.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
 	-lz
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/Makefile.am SimGear-0.3.5-p/simgear/Makefile.am
--- SimGear-0.3.5/simgear/Makefile.am	Mon Mar 29 12:29:11 2004
+++ SimGear-0.3.5-p/simgear/Makefile.am	Mon Mar 29 12:29:46 2004
@@ -18,6 +18,7 @@
 	debug \
 	structure \
 	timing \
+	props \
 	misc \
 	bucket \
 	ephemeris \
@@ -27,7 +28,6 @@
 	math \
 	$(METAR_DIRS) \
 	nasal \
-	props \
 	route \
 	scene \
 	screen \
diff -Naur SimGear-0.3.5/simgear/props/Makefile.am SimGear-0.3.5-p/simgear/props/Makefile.am
--- SimGear-0.3.5/simgear/props/Makefile.am	Mon Mar 29 12:29:11 2004
+++ SimGear-0.3.5-p/simgear/props/Makefile.am	Mon Mar 29 12:30:07 2004
@@ -12,16 +12,16 @@
 	props.cxx \
 	props_io.cxx
 
-noinst_PROGRAMS = props_test
+#noinst_PROGRAMS = props_test
 
-props_test_SOURCES = props_test.cxx
-props_test_LDADD = \
-	$(top_builddir)/simgear/props/libsgprops.la \
-	$(top_builddir)/simgear/xml/libsgxml.la \
-	$(top_builddir)/simgear/misc/libsgmisc.la \
-	$(top_builddir)/simgear/debug/libsgdebug.la \
-	$(top_builddir)/simgear/structure/libsgstructure.la \
-	$(top_builddir)/simgear/timing/libsgtiming.la \
-	-lz
+#props_test_SOURCES = props_test.cxx
+#props_test_LDADD = \
+#	$(top_builddir)/simgear/props/libsgprops.la \
+#	$(top_builddir)/simgear/xml/libsgxml.la \
+#	$(top_builddir)/simgear/misc/libsgmisc.la \
+#	$(top_builddir)/simgear/debug/libsgdebug.la \
+#	$(top_builddir)/simgear/structure/libsgstructure.la \
+#	$(top_builddir)/simgear/timing/libsgtiming.la \
+#	-lz
 
 INCLUDES = -I$(top_srcdir)
diff -Naur SimGear-0.3.5/simgear/io/Makefile.am SimGear-0.3.5-p/simgear/io/Makefile.am
--- SimGear-0.3.5/simgear/io/Makefile.am	Mon Mar 29 12:35:41 2004
+++ SimGear-0.3.5-p/simgear/io/Makefile.am	Mon Mar 29 12:36:50 2004
@@ -75,4 +75,5 @@
 	$(top_builddir)/simgear/debug/libsgdebug.la \
 	$(top_builddir)/simgear/bucket/libsgbucket.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
 	$(base_LIBS) -lz
diff -Naur SimGear-0.3.5/simgear/io/Makefile.am SimGear-0.3.5-p/simgear/io/Makefile.am
--- SimGear-0.3.5/simgear/io/Makefile.am	Mon Mar 29 12:58:11 2004
+++ SimGear-0.3.5-p/simgear/io/Makefile.am	Mon Mar 29 12:52:26 2004
@@ -32,6 +32,9 @@
 	$(top_builddir)/simgear/bucket/libsgbucket.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
 	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
 	-lplibnet -lplibul -lz \
 	$(network_LIBS) \
 	$(base_LIBS)
@@ -44,6 +47,9 @@
 	$(top_builddir)/simgear/bucket/libsgbucket.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
 	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
 	-lplibnet -lplibul -lz \
 	$(network_LIBS) \
 	$(base_LIBS)
@@ -55,6 +61,10 @@
 	$(top_builddir)/simgear/debug/libsgdebug.la \
 	$(top_builddir)/simgear/bucket/libsgbucket.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
 	-lplibnet -lplibul -lz \
 	$(network_LIBS) \
 	$(base_LIBS)
@@ -66,7 +76,11 @@
 	$(top_builddir)/simgear/debug/libsgdebug.la \
 	$(top_builddir)/simgear/bucket/libsgbucket.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
-	$(base_LIBS) -lz
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	$(base_LIBS) -lz -lplibnet -lplibul
 
 decode_binobj_SOURCES = decode_binobj.cxx
 
@@ -76,4 +90,7 @@
 	$(top_builddir)/simgear/bucket/libsgbucket.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
 	$(top_builddir)/simgear/serial/libsgserial.la \
-	$(base_LIBS) -lz
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(base_LIBS) -lz -lplibnet -lplibul
diff -Naur SimGear-0.3.5/simgear/route/Makefile.am SimGear-0.3.5-p/simgear/route/Makefile.am
--- SimGear-0.3.5/simgear/route/Makefile.am	Mon Mar 29 12:58:11 2004
+++ SimGear-0.3.5-p/simgear/route/Makefile.am	Mon Mar 29 12:54:07 2004
@@ -19,6 +19,9 @@
 	$(top_builddir)/simgear/math/libsgmath.la \
 	$(top_builddir)/simgear/debug/libsgdebug.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
 	$(base_LIBS) \
 	-lz
 
@@ -29,5 +32,8 @@
 	$(top_builddir)/simgear/math/libsgmath.la \
 	$(top_builddir)/simgear/debug/libsgdebug.la \
 	$(top_builddir)/simgear/misc/libsgmisc.la \
+	$(top_builddir)/simgear/props/libsgprops.la \
+	$(top_builddir)/simgear/xml/libsgxml.la \
+	$(top_builddir)/simgear/structure/libsgstructure.la \
 	-lz
 	$(base_LIBS)
