diff -Naur SimGear-0.0.14/configure.in SimGear-0.0.14-p/configure.in
--- SimGear-0.0.14/configure.in	Wed Dec 13 20:18:29 2000
+++ SimGear-0.0.14-p/configure.in	Sat Apr 21 23:46:12 2001
@@ -29,7 +29,7 @@
 AC_PROG_MAKE_SET
 AC_PROG_CC
 AC_PROG_CXX
-AC_PROG_RANLIB
+AM_PROG_LIBTOOL
 AC_PROG_INSTALL
 AC_PROG_LN_S
 
diff -Naur SimGear-0.0.14/simgear/bucket/Makefile.am SimGear-0.0.14-p/simgear/bucket/Makefile.am
--- SimGear-0.0.14/simgear/bucket/Makefile.am	Wed Sep 20 23:20:17 2000
+++ SimGear-0.0.14-p/simgear/bucket/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -1,10 +1,10 @@
 includedir = @includedir@/bucket
 
-lib_LIBRARIES = libsgbucket.a
+lib_LTLIBRARIES = libsgbucket.la
 
 include_HEADERS = newbucket.hxx
 
-libsgbucket_a_SOURCES = newbucket.cxx
+libsgbucket_la_SOURCES = newbucket.cxx
 
 # noinst_PROGRAMS = testbucket
 
diff -Naur SimGear-0.0.14/simgear/debug/Makefile.am SimGear-0.0.14-p/simgear/debug/Makefile.am
--- SimGear-0.0.14/simgear/debug/Makefile.am	Wed Sep 20 23:11:22 2000
+++ SimGear-0.0.14-p/simgear/debug/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -2,10 +2,10 @@
 
 EXTRA_DIST = logtest.cxx
 
-lib_LIBRARIES = libsgdebug.a
+lib_LTLIBRARIES = libsgdebug.la
 
 include_HEADERS = debug_types.h logstream.hxx
 
-libsgdebug_a_SOURCES = logstream.cxx
+libsgdebug_la_SOURCES = logstream.cxx
 
 INCLUDES += -I$(top_srcdir)
diff -Naur SimGear-0.0.14/simgear/ephemeris/Makefile.am SimGear-0.0.14-p/simgear/ephemeris/Makefile.am
--- SimGear-0.0.14/simgear/ephemeris/Makefile.am	Tue Oct 10 19:16:21 2000
+++ SimGear-0.0.14-p/simgear/ephemeris/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -1,6 +1,6 @@
 includedir = @includedir@/ephemeris
 
-lib_LIBRARIES = libsgephem.a
+lib_LTLIBRARIES = libsgephem.la
 
 include_HEADERS = \
 	celestialBody.hxx \
@@ -16,7 +16,7 @@
 	uranus.hxx \
 	venus.hxx
 
-libsgephem_a_SOURCES = \
+libsgephem_la_SOURCES = \
 	celestialBody.cxx \
 	ephemeris.cxx \
 	jupiter.cxx \
diff -Naur SimGear-0.0.14/simgear/io/Makefile.am SimGear-0.0.14-p/simgear/io/Makefile.am
--- SimGear-0.0.14/simgear/io/Makefile.am	Mon Oct  2 23:43:01 2000
+++ SimGear-0.0.14-p/simgear/io/Makefile.am	Sat Apr 21 23:47:02 2001
@@ -1,6 +1,6 @@
 includedir = @includedir@/io
 
-lib_LIBRARIES = libsgio.a
+lib_LTLIBRARIES = libsgio.la
 
 include_HEADERS = \
 	iochannel.hxx \
@@ -8,7 +8,7 @@
 	sg_serial.hxx \
 	sg_socket.hxx
 
-libsgio_a_SOURCES = \
+libsgio_la_SOURCES = \
 	iochannel.cxx \
 	sg_file.cxx \
 	sg_serial.cxx \
@@ -21,5 +21,6 @@
 socktest_SOURCES = socktest.cxx
 
 socktest_LDADD = \
-	$(top_builddir)/simgear/io/libsgio.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a 
\ No newline at end of file
+	$(top_builddir)/simgear/io/libsgio.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/serial/libsgserial.la
diff -Naur SimGear-0.0.14/simgear/magvar/Makefile.am SimGear-0.0.14-p/simgear/magvar/Makefile.am
--- SimGear-0.0.14/simgear/magvar/Makefile.am	Wed Sep 20 23:20:23 2000
+++ SimGear-0.0.14-p/simgear/magvar/Makefile.am	Sat Apr 21 23:47:06 2001
@@ -1,10 +1,10 @@
 includedir = @includedir@/magvar
 
-lib_LIBRARIES = libsgmagvar.a
+lib_LTLIBRARIES = libsgmagvar.la
 
 include_HEADERS = magvar.hxx
 
-libsgmagvar_a_SOURCES = \
+libsgmagvar_la_SOURCES = \
 	coremag.hxx coremag.cxx \
 	magvar.cxx
 
@@ -12,6 +12,6 @@
 
 testmagvar_SOURCES = testmagvar.cxx
 
-testmagvar_LDADD = $(top_builddir)/simgear/magvar/libsgmagvar.a
+testmagvar_LDADD = $(top_builddir)/simgear/magvar/libsgmagvar.la
 
 INCLUDES += -I$(top_srcdir)
diff -Naur SimGear-0.0.14/simgear/math/Makefile.am SimGear-0.0.14-p/simgear/math/Makefile.am
--- SimGear-0.0.14/simgear/math/Makefile.am	Mon Dec  4 16:04:49 2000
+++ SimGear-0.0.14-p/simgear/math/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -6,7 +6,7 @@
 ZLIB_INCL = -I$(top_srcdir)/src/zlib
 endif
 
-lib_LIBRARIES = libsgmath.a
+lib_LTLIBRARIES = libsgmath.la
 
 include_HEADERS = \
 	fg_memory.h \
@@ -22,7 +22,7 @@
 
 EXTRA_DIST = linintp2.h linintp2.inl sphrintp.h sphrintp.inl
 
-libsgmath_a_SOURCES = \
+libsgmath_la_SOURCES = \
 	interpolater.cxx \
 	leastsqs.cxx \
 	polar3d.cxx \
diff -Naur SimGear-0.0.14/simgear/metar/Makefile.am SimGear-0.0.14-p/simgear/metar/Makefile.am
--- SimGear-0.0.14/simgear/metar/Makefile.am	Wed Sep 20 23:12:22 2000
+++ SimGear-0.0.14-p/simgear/metar/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -1,8 +1,8 @@
 includedir = @includedir@/metar
 
-lib_LIBRARIES = libsgmetar.a
+lib_LTLIBRARIES = libsgmetar.la
 
-libsgmetar_a_SOURCES = \
+libsgmetar_la_SOURCES = \
 	Antoi.cpp Charcmp.cpp Dcdmetar.cpp Dcdmtrmk.cpp Drvmetar.c \
 	Fracpart.cpp Local.h Metar.h \
 	MetarReport.cpp MetarReport.h \
diff -Naur SimGear-0.0.14/simgear/misc/Makefile.am SimGear-0.0.14-p/simgear/misc/Makefile.am
--- SimGear-0.0.14/simgear/misc/Makefile.am	Wed Sep 20 23:12:30 2000
+++ SimGear-0.0.14-p/simgear/misc/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -6,7 +6,7 @@
 ZLIB_INCL = -I$(top_srcdir)/src/zlib
 endif
 
-lib_LIBRARIES = libsgmisc.a
+lib_LTLIBRARIES = libsgmisc.la
 
 include_HEADERS = \
 	fgpath.hxx \
@@ -17,7 +17,7 @@
 	texcoord.hxx \
 	zfstream.hxx
 
-libsgmisc_a_SOURCES = \
+libsgmisc_la_SOURCES = \
 	fgpath.cxx \
 	fgstream.cxx \
 	props.cxx \
diff -Naur SimGear-0.0.14/simgear/route/Makefile.am SimGear-0.0.14-p/simgear/route/Makefile.am
--- SimGear-0.0.14/simgear/route/Makefile.am	Mon Oct 23 20:05:24 2000
+++ SimGear-0.0.14-p/simgear/route/Makefile.am	Sat Apr 21 23:47:28 2001
@@ -1,10 +1,10 @@
 includedir = @includedir@/route
 
-lib_LIBRARIES = libsgroute.a
+lib_LTLIBRARIES = libsgroute.la
 
 include_HEADERS = route.hxx waypoint.hxx
 
-libsgroute_a_SOURCES = \
+libsgroute_la_SOURCES = \
 	route.cxx \
 	waypoint.cxx
 
@@ -15,13 +15,17 @@
 waytest_SOURCES = waytest.cxx
 
 waytest_LDADD = \
-	$(top_builddir)/simgear/route/libsgroute.a \
-	$(top_builddir)/simgear/math/libsgmath.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a
+	$(top_builddir)/simgear/route/libsgroute.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la -lz \
+	$(top_builddir)/simgear/xml/libsgxml.la
 
 routetest_SOURCES = routetest.cxx
 
 routetest_LDADD = \
-	$(top_builddir)/simgear/route/libsgroute.a \
-	$(top_builddir)/simgear/math/libsgmath.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a
\ No newline at end of file
+	$(top_builddir)/simgear/route/libsgroute.la \
+	$(top_builddir)/simgear/math/libsgmath.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la \
+	$(top_builddir)/simgear/misc/libsgmisc.la -lz \
+	$(top_builddir)/simgear/xml/libsgxml.la
diff -Naur SimGear-0.0.14/simgear/screen/Makefile.am SimGear-0.0.14-p/simgear/screen/Makefile.am
--- SimGear-0.0.14/simgear/screen/Makefile.am	Wed Sep 20 23:12:39 2000
+++ SimGear-0.0.14-p/simgear/screen/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -1,10 +1,10 @@
 includedir = @includedir@/screen
 
-lib_LIBRARIES = libsgscreen.a
+lib_LTLIBRARIES = libsgscreen.la
 
 include_HEADERS = screen-dump.hxx
 
-libsgscreen_a_SOURCES = \
+libsgscreen_la_SOURCES = \
 	GLBitmaps.cxx GLBitmaps.h \
 	screen-dump.cxx \
 	win32-printer.h
diff -Naur SimGear-0.0.14/simgear/serial/Makefile.am SimGear-0.0.14-p/simgear/serial/Makefile.am
--- SimGear-0.0.14/simgear/serial/Makefile.am	Wed Sep 20 23:20:31 2000
+++ SimGear-0.0.14-p/simgear/serial/Makefile.am	Sat Apr 21 23:47:35 2001
@@ -2,16 +2,16 @@
 
 noinst_PROGRAMS = testserial
 
-lib_LIBRARIES = libsgserial.a
+lib_LTLIBRARIES = libsgserial.la
 
 include_HEADERS = serial.hxx
 
-libsgserial_a_SOURCES = serial.cxx
+libsgserial_la_SOURCES = serial.cxx
 
 testserial_SOURCES = testserial.cxx
 
 testserial_LDADD = \
-	$(top_builddir)/simgear/serial/libsgserial.a \
-	$(top_builddir)/simgear/debug/libsgdebug.a
+	$(top_builddir)/simgear/serial/libsgserial.la \
+	$(top_builddir)/simgear/debug/libsgdebug.la
 
 INCLUDES += -I$(top_srcdir)
diff -Naur SimGear-0.0.14/simgear/sky/Makefile.am SimGear-0.0.14-p/simgear/sky/Makefile.am
--- SimGear-0.0.14/simgear/sky/Makefile.am	Wed Sep 20 23:12:56 2000
+++ SimGear-0.0.14-p/simgear/sky/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -1,6 +1,6 @@
 includedir = @includedir@/sky
 
-lib_LIBRARIES = libsgsky.a
+lib_LTLIBRARIES = libsgsky.la
 
 include_HEADERS = \
 	cloud.hxx \
@@ -11,7 +11,7 @@
 	sphere.hxx \
 	stars.hxx
 
-libsgsky_a_SOURCES = \
+libsgsky_la_SOURCES = \
 	cloud.cxx \
 	dome.cxx \
 	moon.cxx \
diff -Naur SimGear-0.0.14/simgear/timing/Makefile.am SimGear-0.0.14-p/simgear/timing/Makefile.am
--- SimGear-0.0.14/simgear/timing/Makefile.am	Thu Nov 30 18:10:16 2000
+++ SimGear-0.0.14-p/simgear/timing/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -1,6 +1,6 @@
 includedir = @includedir@/timing
 
-lib_LIBRARIES = libsgtiming.a
+lib_LTLIBRARIES = libsgtiming.la
 
 include_HEADERS = \
 	geocoord.h \
@@ -9,7 +9,7 @@
 	timestamp.hxx \
 	timezone.h
 
-libsgtiming_a_SOURCES = \
+libsgtiming_la_SOURCES = \
 	geocoord.cxx \
 	lowleveltime.cxx \
 	sg_time.cxx \
diff -Naur SimGear-0.0.14/simgear/xgl/Makefile.am SimGear-0.0.14-p/simgear/xgl/Makefile.am
--- SimGear-0.0.14/simgear/xgl/Makefile.am	Wed Sep 20 23:13:01 2000
+++ SimGear-0.0.14-p/simgear/xgl/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -1,9 +1,9 @@
 includedir = @includedir@/xgl
 
-lib_LIBRARIES = libsgxgl.a
+lib_LTLIBRARIES = libsgxgl.la
 
 include_HEADERS = xgl.h
 
-libsgxgl_a_SOURCES = xgl.c xglUtils.c
+libsgxgl_la_SOURCES = xgl.c xglUtils.c
 
 INCLUDES += -I$(top_srcdir)
diff -Naur SimGear-0.0.14/simgear/xml/Makefile.am SimGear-0.0.14-p/simgear/xml/Makefile.am
--- SimGear-0.0.14/simgear/xml/Makefile.am	Wed Sep 20 23:13:09 2000
+++ SimGear-0.0.14-p/simgear/xml/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -4,7 +4,7 @@
 XLIB_INCL = -I$(top_srcdir)/src/zlib
 endif
 
-lib_LIBRARIES = libsgxml.a
+lib_LTLIBRARIES = libsgxml.la
 
 include_HEADERS = \
 	easyxml.hxx
@@ -12,7 +12,7 @@
 noinst_HEADERS = \
 	xmltok_impl.c xmltok_ns.c
 
-libsgxml_a_SOURCES = \
+libsgxml_la_SOURCES = \
         asciitab.h \
 	easyxml.cxx \
 	hashtable.h hashtable.c \
diff -Naur SimGear-0.0.14/simgear/zlib/Makefile.am SimGear-0.0.14-p/simgear/zlib/Makefile.am
--- SimGear-0.0.14/simgear/zlib/Makefile.am	Sat Feb 19 03:22:56 2000
+++ SimGear-0.0.14-p/simgear/zlib/Makefile.am	Sat Apr 21 23:46:12 2001
@@ -3,11 +3,11 @@
 EXTRA_DIST = ChangeLog INDEX Make_vms.com Makefile.riscos README \
 	algorithm.txt descrip.mms example.c maketree.c minigzip.c zlib.3
 
-lib_LIBRARIES = libz.a
+lib_LTLIBRARIES = libz.la
 
 include_HEADERS = zlib.h zconf.h
 
-libz_a_SOURCES = \
+libz_la_SOURCES = \
 	adler32.c \
 	compress.c \
 	crc32.c \
--- SimGear-0.0.14/simgear/Makefile.am	Tue Sep 19 18:19:43 2000
+++ SimGear-0.0.14-p/simgear/Makefile.am	Sun Apr 22 00:16:15 2001
@@ -28,17 +28,17 @@
 	bucket \
 	debug \
 	ephemeris \
+	$(SERIAL_DIRS) \
 	io \
 	magvar \
 	math \
 	$(METAR_DIRS) \
 	misc \
+	xml \
 	route \
 	screen \
-	$(SERIAL_DIRS) \
 	sky \
 	timing \
 	xgl \
-	xml \
 	$(ZLIB_DIRS)
 
